src/ape/pytest/runners.py

```diff
@@ -272,3 +248,37 @@ def pytest_unconfigure(self):
         self.chain_manager.contracts.clear_local_caches()
         self.gas_tracker.session_gas_report = None
         self.coverage_tracker.reset()
+
+
+def _insert_isolation_fixtures(item):
+    # Abstracted for testing purposes.
+    fixture_map = item.session._fixturemanager._arg2fixturedefs
+    scopes = [
+        definition.scope
+        for name, definitions in fixture_map.items()
+        if name in item.fixturenames
+        for definition in definitions
+    ]
+
+    for scope in ("session", "package", "module", "class"):
+        # iterate through scope levels and insert the isolation fixture
+        # prior to the first fixture with that scope
+        try:
+            idx = scopes.index(scope)  # will raise ValueError if `scope` not found
+        except ValueError:
+            continue
+
+        name = f"_{scope}_isolation"
+
+        # Don't let isolation fixtures get added more than once!
+        if name not in item.fixturenames:
```

Comment:
this is a new check , I wonder if this resolved the culprit. Basically just don't allow isolation fixture from getting added more than once.

I am going to check if this resolves https://github.com/ApeWorX/ape/issues/1287